### PR TITLE
Parse attributes in `#if`

### PR DESCRIPTION
**Pull request by ahoppen**
Fixes apple/swift-syntax#1119
rdar://103048943

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SwiftParser/Lookahead.swift [hunk 1]
```diff
@@ -207,42 +207,35 @@ extension Parser.Lookahead {
     return true
   }
 
+  /// Tries consuming a `#if` directive that only contains attributes.
+  /// Returns `true` if that was successful and `false` if
+  ///  - we are not at a valid `#if` directive (e.g. `#elseif` is missing)
```
Comment (bnbarham) [inline Sources/SwiftParser/Lookahead.swift:212]:
```suggestion
  ///  - we are not at a valid `#if` directive (e.g. `#endif` is missing)
```

File: Sources/SwiftParser/Lookahead.swift [hunk 2]
```diff
@@ -207,42 +207,35 @@ extension Parser.Lookahead {
...
+  ///  - the directive contained non-attributes
+  ///  - the directive did not contain any attributes
   mutating func consumeIfConfigOfAttributes() -> Bool {
-    while true {
-      // #if / #else / #elseif
-      self.consumeAnyToken()
+    var didSeeAnyAttributes = false
+    var poundIfLoopProgress = LoopProgressCondition()
+    repeat {
+      self.consume(ifAny: [.poundIfKeyword, .poundElseKeyword, .poundEndifKeyword])
```
Comment (rintaro) [inline Sources/SwiftParser/Lookahead.swift:219]: You meant `poundElseifKeyword`?
```suggestion
      assert(self.at(any: [.poundIfKeyword, .poundElseifKeyword, .poundElseKeyword])
      consumeAnyToken()
```

File: Sources/SwiftParser/Declarations.swift [hunk 3]
```diff
@@ -120,6 +120,13 @@ extension TokenConsumer {
       return false
     }
   }
+
+  mutating func atStartOfIfConfigOfAttributes() -> Bool {
```
Comment (rintaro) [inline Sources/SwiftParser/Declarations.swift:124]: Unused?

Comment (ahoppen) [reply Sources/SwiftParser/Declarations.swift:124]: Oh, yes. Thanks

File: Sources/SwiftParser/Lookahead.swift (hunk 2 above)
Comment (ahoppen) [reply Sources/SwiftParser/Lookahead.swift:219]: Yes, thanks

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by rintaro — APPROVED** [review]

**Pull request merged**